openstack/barbican/templates/barbican-nanny-deployment.yaml

```diff
@@ -0,0 +1,112 @@
+{{- if .Values.barbican_nanny.enabled }}
+{{- if .Values.barbican_nanny.db_secret_move.enabled }}
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: barbican-nanny
+  labels:
+    app: {{ template "name" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+    type: nanny
+    component: barbican
+spec:
+  replicas: {{ .Values.barbican_nanny.replicas }}
+  revisionHistoryLimit: {{ .Values.pod.lifecycle.upgrades.deployments.revisionHistory }}
+  strategy:
+    type: {{ .Values.pod.lifecycle.upgrades.deployments.podReplacementStrategy }}
+    {{ if eq .Values.pod.lifecycle.upgrades.deployments.podReplacementStrategy "RollingUpdate" }}
+    rollingUpdate:
+      maxUnavailable: {{ .Values.pod.lifecycle.upgrades.deployments.rollingUpdate.maxUnavailable }}
+      maxSurge: {{ .Values.pod.lifecycle.upgrades.deployments.rollingUpdate.maxSurge }}
+    {{ end }}
+  selector:
+    matchLabels:
+      app: barbican-nanny
+  template:
+    metadata:
+      labels:
+        app: barbican-nanny
+        name: barbican
+        alert-tier: os
+        alert-service: barbican
+        {{- include "utils.topology.pod_label" . | indent 8 }}
+      annotations:
+        chart-version: {{.Chart.Version}}
+        checksum/etc-configmap.conf: {{ include (print $.Template.BasePath "/etc-configmap.yaml") . | sha256sum }}
+        secrets-hash: {{ include (print $.Template.BasePath "/secrets.yaml") . | sha256sum }}
```

Comment:
It would make sense to also add the hash of the `barbican-scripts` configmap here, because any change to the script should re-create this pod.